larq/quantizers_test.py

```diff
@@ -69,3 +69,32 @@ def approx_sign_grad(x):
         activation = lq.quantizers.approx_sign(tf_x)
     grad = tape.gradient(activation, tf_x)
     np.testing.assert_allclose(grad.numpy(), approx_sign_grad(x))
+
+
+@pytest.mark.skipif(not tf.executing_eagerly(), reason="requires eager execution")
+def test_magnitude_aware_sign(self):
+    if tf.executing_eagerly():
+        return
+
+    a = np.random.uniform(-2, 2, (3, 2, 2, 3))
+    x = tf.constant(a, dtype=tf.float32)
+    y = lq.quantizers.magnitude_aware_sign(x)
+    grad = tf.gradients(y, x)
+
+    assert y.shape == x.shape
+
+    # check sign
+    np.testing.assert_allclose(tf.sign(y).numpy(), np.sign(a))
+
+    scale_vector = [np.mean(np.reshape(np.abs(a[:, :, :, i]))) for i in range(3)]
+
+    # check magnitude
+    np.testing.assert_allclose(
+        tf.reduce_mean(tf.abs(y), axis=[0, 1, 2]).numpy(),
```

Comment:
```suggestion
        lq.quantizers.magnitude_aware_sign(x).numpy(),
```

Shouldn't this test the quantizer?

Reply:
This tests the magnitude of the quantized vector